src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -24,119 +63,406 @@ PhysicsRobot::~PhysicsRobot()
+    // so we do not need to rotate the points to match the orientation of the robot.
+    const unsigned int num_vertices              = 4;
+    b2Vec2 dribbler_shape_vertices[num_vertices] = {
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS, DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS - dribbler_depth,
+                         DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS - dribbler_depth,
+                         -DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS, -DRIBBLER_WIDTH_METERS / 2.0))};
+    b2PolygonShape* dribbler_shape = new b2PolygonShape();
+    dribbler_shape->Set(dribbler_shape_vertices, num_vertices);
+    robot_dribbler_fixture_def.shape = dribbler_shape;
+    robot_body->CreateFixture(&robot_dribbler_fixture_def);
+}
+
+void PhysicsRobot::setupChickerFixture(const Robot& robot, double total_chicker_depth,
+                                       double chicker_thickness)
+{
+    b2FixtureDef robot_chicker_fixture_def;
+    // We want perfect damping to help us keep the ball when dribbling
+    robot_chicker_fixture_def.restitution = 0.0;
+    // We want lots of friction for when the ball is being dribbled so it stays controlled
+    robot_chicker_fixture_def.friction = 1.0;
+    // The chicker has no mass in simulation. Mass is already accounted for by the robot
+    // body
+    robot_chicker_fixture_def.density = 0.0;
+    robot_chicker_fixture_def.userData =
+        new PhysicsObjectUserData({PhysicsObjectType::ROBOT_CHICKER, this});
+
+    // Box2D requires that polygon vertices are specified in counter-clockwise order.
+    // The fixture shape is added relative to the body in its local coordinate frame,
+    // so we do not need to rotate the points to match the orientation of the robot.
+    const unsigned int num_vertices             = 4;
+    b2Vec2 chicker_shape_vertices[num_vertices] = {
+        createVec2(
+            Point(DIST_TO_FRONT_OF_ROBOT_METERS - total_chicker_depth + chicker_thickness,
+                  DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS - total_chicker_depth,
+                         DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(Point(DIST_TO_FRONT_OF_ROBOT_METERS - total_chicker_depth,
+                         -DRIBBLER_WIDTH_METERS / 2.0)),
+        createVec2(
+            Point(DIST_TO_FRONT_OF_ROBOT_METERS - total_chicker_depth + chicker_thickness,
+                  -DRIBBLER_WIDTH_METERS / 2.0))};
+
+    b2PolygonShape* chicker_shape = new b2PolygonShape();
+    chicker_shape->Set(chicker_shape_vertices, num_vertices);
+    robot_chicker_fixture_def.shape = chicker_shape;
+    robot_body->CreateFixture(&robot_chicker_fixture_def);
 }
 
-void PhysicsRobot::setupRobotPhysicsBody(const Robot& robot)
+b2PolygonShape* PhysicsRobot::getMainRobotBodyShape(const Robot& robot,
```

Comment:
To add to what I said above, this would be good to have in a Model class so this class isn't too cluttered. It kind of has multiple responsibilities right now I think

Reply:
I agree. I created a `PhysicsRobotModel` class that handles creating the polygon shapes.